### ISSUE TITLE
WinAppDriver doesn't detect elements after a window reload

### ISSUE THREAD

**Issue by pradeipp** [microsoft/WinAppDriver#1225]
I'm automating a simple workflow in Microsoft Word using WinAppDriver. In it, I use a custom word addin to generate a  document template that's used by our clients. 
The workflow goes like this: 

**Open word** -> **Open Addin** -> **Fill dialog box and click ok** -> **_**word reloads with the newly generated document.**_

Everything works fine before the dialog box, But as soon as I click on the finish button, word reloads, and the old session possibly expires. 

Now I do use the session refreshing mechanism as provided by WinAppDriver authors. It involves creating a new ```WindowsDriver``` session by extracting the ```NativeWindowHandle``` of the MsWord window by searching from the **root,** element of the desktop. 
But this newly created fresh MsWord session still doesn't detect any elements. 

And to add to the confusion the new session detects some parts of the application(header pane buttons) and doesn't detect top level items.

What could be causing this issue?

**Comment by anunay1** [comment]
Does inspect show the top level elements

**Comment by pradeipp** [comment]
@anunay1 yup.. detects all the elments. Checked on inspect.exe, WAD UI recorder and even FlaUI inspect

**Comment by anunay1** [comment]
> @anunay1 yup.. detects all the elments. Checked on inspect.exe, WAD UI recorder and even FlaUI inspect

Then it should detect the top level window, try adding some delay.

**Comment by pradeipp** [comment]
HI @anunay1 , Sorry I meant the top row elements of MsWord like the tabs and ribbons which have things like font settings, table settings, etc. 
The top level window does get detected. Both through the window handle method and through the desktop->FindElementByClassName method. The problem is that it just doesn't detect anything inside this new instance of the window.

**Comment by anunay1** [comment]
can you share the code?

**Comment by anunay1** [comment]
(The problem is that it just doesn't detect anything inside this new instance of the window.) Is inspect able to detect.

**Comment by pradeipp** [comment]
Hi @anunay1  I've updated the post with the code. I've tried to make it as easily readable as possible. The reason why all FindElements methods fail after the window reload is what we want to get at. And finally be able to click on the tab, save and check title bar until it has a "saved" text. Thank you for lending a helping hand, appreciate it.

**Comment by anunay1** [comment]
Sorry dude, it has a lot of global functions, I cannot make out. 
Can you check this issue #894

**Comment by pradeipp** [comment]
@anunay1  that post didn't help much either. I also get this error when trying to get multiple elements in the new refreshed window, which I forgot to mention:

``` bash
POST /session/4F701775-725B-4D68-94DB-8A8BFB1D2281/elements HTTP/1.1
Accept: application/json, image/png
Content-Length: 36
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.0 (.net windows)

{"using":"xpath","value":"//Button"}

### SearchMultipleElements exception: Error HRESULT E_FAIL has been returned from a call to a COM component.

```

**Comment by pradeipp** [comment]
Sounds weird but [->this<-](https://github.com/microsoft/WinAppDriver/issues/529#issuecomment-445200485) seemed to do the trick. Not sure how a seemingly useless implicitwait definition could ruin the functionality of WinAppDriver in such a drastic way(I'd seen [->this<-](https://github.com/microsoft/WinAppDriver/issues/682) as well as [->this<-](https://github.com/microsoft/WinAppDriver/issues/196) issue beforehand about the implicit wait being inefficient). 
But it is what it is I guess. 

Here's what happened in my case:
In the session definition, which goes like this:

``` C#
    var options = new AppiumOptions();
    options.AddAdditionalCapability("app", "Root");
    options.AddAdditionalCapability("deviceName", PcName);
    DesktopSession = new WindowsDriver<WindowsElement>(new Uri(WinAppDriverUrl), options, 
    TimeSpan.FromSeconds(timeOutInSeconds));

    Assert.IsNotNull(DesktopSession);
    DesktopSession.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(1.5);
```

I'd removed the last line(that too just in this desktop session definition) and it somehow broke the whole functionality.  Put the thing there again and it seems to work. At least for now. 
I'll leave the thread open as I'm not sure if this resolution is even it though.
